app/src/main/java/com/hedvig/app/feature/offer/ui/SignMethodExt.kt

```diff
@@ -0,0 +1,34 @@
+package com.hedvig.app.feature.offer.ui
+
+import android.widget.Button
+import androidx.core.content.ContextCompat
+import com.google.android.material.button.MaterialButton
+import com.hedvig.android.owldroid.type.SignMethod
+import com.hedvig.app.R
+import java.lang.IllegalArgumentException
+
+fun SignMethod.checkoutTextRes() = when (this) {
+    SignMethod.SWEDISH_BANK_ID -> R.string.OFFER_SIGN_BUTTON
+    SignMethod.SIMPLE_SIGN -> R.string.OFFER_CHECKOUT_BUTTON
+    SignMethod.APPROVE_ONLY -> R.string.OFFER_APPROVE_CHANGES
+    SignMethod.NORWEGIAN_BANK_ID, // Deprecated
+    SignMethod.DANISH_BANK_ID, // Deprecated
+    SignMethod.UNKNOWN__ -> throw IllegalArgumentException("Could not parse checkout string for $this")
```

Comment:
This `throw` makes it so that we need to guard against calling this function on a higher layer, which makes it a bit dangerous 😕 We also do not have this guard in place in the `OfferActivity`, meaning that introducing a new sign method would make any app user opening the Offer Screen crash instead of having a more graceful degradation of their experience.

Reply:
Very true, good catch. Fixing that.